lib/adapters/http/http.js

```diff
@@ -73,7 +73,7 @@ function getHost(name, opts) {
     uri.path = parts.join('/');
     opts = opts || {};
     opts = utils.clone(opts);
-    uri.headers = opts.headers || {};
+    uri.headers = opts.headers || (opts.ajax && opts.ajax.headers) || {};
 
     if (opts.auth || uri.auth) {
       var nAuth = opts.auth || uri.auth;
```

Comment:
^ This is where `username` and `password` come from. Is there any way to test this in your components test?